### ISSUE TITLE
 Integrity constraint violation: 1452 after delete website

### ISSUE THREAD

**Issue by davirs** [magento/magento2#30503]
### Preconditions (*)

1. Magento 2.4
2. Two configured websites
3. **First** website with 5 store views and the **second** website with one store view

### Steps to reproduce (*)

1. Delete the 4 stoweviews from **first** website (in my case I mantein website id 5)
2. Set the layout for the first store view to see correctly the frontend
3. Delete the **second** website
4. Check if there is one one website and one store view.
![image](https://user-images.githubusercontent.com/4355062/96172177-68ef6900-0efc-11eb-8237-11d411676e49.png)

5. Go to catalog/product
6. Add a new configurable product(for example) and register the product filling everything as default. You can insert anything until finish.
7. Click at the save buttom

### Expected result (*)

1. The expected result is register the product normally without problems

### Actual result (*)

1. `SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`loja_magento`.`catalog_product_website`, CONSTRAINT `CATALOG_PRODUCT_WEBSITE_WEBSITE_ID_STORE_WEBSITE_WEBSITE_ID` FOREIGN KEY (`website_id`) REFERENCES `store_website` (`website_id`) ON DELET), query was: INSERT INTO `catalog_product_website` (`product_id`,`website_id`) VALUES (?, ?)`

![image](https://user-images.githubusercontent.com/4355062/96172290-950aea00-0efc-11eb-8209-a595bf66f214.png)

2. I can't create or edit products and anything related with "website_id"

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by davirs** [comment]
Complementing this issue, I discoved if I have only one website and only one store view after the simulation described above, magento will return that error and inside product configuration not shows "Product in Websites" section:

![image](https://user-images.githubusercontent.com/4355062/96280700-ad3b4180-0fae-11eb-9f3f-9ced39efcf32.png)

If I insert one more store view for the currently website, magento shows "Product in Websites " section at product configuration and works fine with out that error.

![image](https://user-images.githubusercontent.com/4355062/96281101-3e121d00-0faf-11eb-9d46-ee66193a3c33.png)

![image](https://user-images.githubusercontent.com/4355062/96281247-71ed4280-0faf-11eb-864f-4c630a5ea4e0.png)

But, I don't need to use more than one store view... 
Can someone help me if any simple solutions while this problem is processed?

**Comment by engcom-Alfa** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Alfa** [comment]
Hi @davirs .

Unfortunately, we are not able to reproduce this issue on a fresh 2.4-develop instance following by your scenario.
Everything works as expected.

**Actual Result:** :heavy_check_mark: A product is saved without any errors.

![Screenshot from 2020-11-16 13-42-00](https://user-images.githubusercontent.com/51679138/99248701-8cfedc80-2811-11eb-9eae-9ec62403329b.png)

@davirs Are you able to reproduce this issue on a clean Magento 2.4-develop instance?
Thanks!

**Comment by davirs** [comment]
@engcom-Alfa The problem exists. If I delete the first website (default magento) and maintein another websites that error will appears. To bypass this problem I only renamed this default website and deleted all another websites. For example:
![image](https://user-images.githubusercontent.com/4355062/99289375-9d1bb980-281b-11eb-8c98-13a2f0821949.png)


Another ploblem is related with Default Store View inside the Store Information. 
Basically you can delete websites but magento is not enought smart to automatically set inside "Default Store View" the last ID remaining and the same error will appears inside products.
Example, I deleted store view en6, but inside "Default Store View" this store still is apointed.

![image](https://user-images.githubusercontent.com/4355062/99289260-7a89a080-281b-11eb-90ff-985a6e8c8059.png)

**Comment by manugentoo** [comment]
Experiencing this now +1 with @davirs following this thread.

**Comment by davirs** [comment]
I dont found a solution, but I really think it's a magento bug.
You don't are able to delede ID 0. For example, you can have 10 stores, but if you delete storefront id 0 this probleem will occurs.

Temporary Solution: Dont revome ID 0, rename it and all things still work fine
